### PR TITLE
Add benchmark for iter_torch_batch/to_tf APIs

### PR DESCRIPTION
**Pull request by jianoaix**
## Why are these changes needed?
These are core APIs in Datasets to consume tensor content held in a Dataset. We should add benchmark tests to make sure it works efficiently. 

## Related issue number
#31446

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jianoaix** [comment]
It's quite expensive to shuffle those image objects so I set to small-ish batch size in test, to make it able to finish in reasonable amount of time:
```
Running case: iter-torch-batches-default
iter_torch_batches done, block_format: BlockFormat.ARROW num_rows: 1624 num_blocks: 8 num_batches: 7
Result of case iter-torch-batches-default: {'time': 15.116176945999541}
Running case: to-tf-default
Result of case to-tf-default: {'time': 5.400960197000131}
Running case: iter-torch-batches-16
iter_torch_batches done, block_format: BlockFormat.ARROW num_rows: 1624 num_blocks: 8 num_batches: 102
Result of case iter-torch-batches-16: {'time': 10.007361148999735}
Running case: to-tf-16
Result of case to-tf-16: {'time': 0.024630216999867116}
Running case: iter-torch-batches-32
iter_torch_batches done, block_format: BlockFormat.ARROW num_rows: 1624 num_blocks: 8 num_batches: 51
Result of case iter-torch-batches-32: {'time': 9.952196469999762}
Running case: to-tf-32
Result of case to-tf-32: {'time': 0.033565447999535536}
Running case: iter-torch-batches-shuffle-16-16
iter_torch_batches done, block_format: BlockFormat.ARROW num_rows: 1624 num_blocks: 8 num_batches: 102
Result of case iter-torch-batches-shuffle-16-16: {'time': 220.0881187949999}
Running case: to-tf-shuffle-16-16
Result of case to-tf-shuffle-16-16: {'time': 0.027030580999962694}
Running case: iter-torch-batches-shuffle-16-32
iter_torch_batches done, block_format: BlockFormat.ARROW num_rows: 1624 num_blocks: 8 num_batches: 102
Result of case iter-torch-batches-shuffle-16-32: {'time': 233.4010537590002}
Running case: to-tf-shuffle-16-32
Result of case to-tf-shuffle-16-32: {'time': 0.025749791999260196}
Running case: iter-torch-batches-shuffle-32-32
iter_torch_batches done, block_format: BlockFormat.ARROW num_rows: 1624 num_blocks: 8 num_batches: 51
Result of case iter-torch-batches-shuffle-32-32: {'time': 181.6337573180008}
Running case: to-tf-shuffle-32-32
Result of case to-tf-shuffle-32-32: {'time': 0.02495413799988455}
Running case: iter-torch-batches-shuffle-32-64
iter_torch_batches done, block_format: BlockFormat.ARROW num_rows: 1624 num_blocks: 8 num_batches: 51
Result of case iter-torch-batches-shuffle-32-64: {'time': 215.00159258500025}
Running case: to-tf-shuffle-32-64
Result of case to-tf-shuffle-32-64: {'time': 0.02457433299969125}
Finish benchmark: iter-tensor-batches

real    17m49.957s
user    7m18.663s
sys     7m29.159s
```

**Review by c21 — APPROVED** [review]

File: release/nightly_tests/dataset/iter_tensor_batches_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+
+def run_iter_tensor_batches_benchmark(benchmark: Benchmark):
+    ds = ray.data.read_images(
+        "s3://anonymous@air-example-data-2/1G-image-data-synthetic-raw"
+    ).fully_executed()
+
+    # Repartition both to align the block sizes so we can zip them.
+    ds = ds.repartition(ds.num_blocks())
+    label = ray.data.range_tensor(ds.count()).repartition(ds.num_blocks())
```
[the remark this answers is not in the document]
Comment (jianoaix) [reply release/nightly_tests/dataset/iter_tensor_batches_benchmark.py:65]: The iter_torch_batch/tf requires a label column which doesn't exist in the image file, so we have to add one here.

File: release/release_tests.yaml [hunk 2]
```diff
@@ -4339,6 +4339,24 @@
     type: sdk_command
     file_manager: sdk
 
+- name: iter_tensor_batches_benchmark_single_node
+  group: core-dataset-tests
```
[the remark this answers is not in the document]
Comment (jianoaix) [reply release/release_tests.yaml:4343]: Done.

**Comment by c21** [comment]
@ericl - just to provide more context, benchmark for iter_batches is already added before (https://github.com/ray-project/ray/pull/31304).

**Review by ericl — APPROVED** [review]

**Pull request merged**